extensions/kubernetes-client/deployment/src/main/java/io/quarkus/kubernetes/client/deployment/DevServicesKubernetesProcessor.java

```diff
@@ -339,32 +316,28 @@ private class RunningContainer implements ContainerState {
         private final InspectContainerResponse containerInfo;
 
         private final ContainerAddress containerAddress;
+        private final Flavor flavor;
 
-        public RunningContainer(DockerClient dockerClient, ContainerAddress containerAddress) {
+        public RunningContainer(DockerClient dockerClient, ContainerAddress containerAddress, Flavor flavor) {
             this.dockerClient = dockerClient;
             this.containerAddress = containerAddress;
             this.containerInfo = dockerClient.inspectContainerCmd(getContainerId()).exec();
+            this.flavor = flavor;
         }
 
         public Map<String, String> getKubeconfig() {
-            var image = getContainerInfo().getConfig().getImage();
-            if (image.contains("rancher/k3s")) {
-                return getKubernetesClientConfigFromKubeConfig(
+            return switch (flavor) {
+                case k3s -> getKubernetesClientConfigFromKubeConfig(
                         KubeConfigUtils.parseKubeConfig(KubeConfigUtils.replaceServerInKubeconfig(containerAddress.getUrl(),
                                 getFileContentFromContainer(K3S_KUBECONFIG))));
-            } else if (image.contains("kindest/node")) {
-                return getKubernetesClientConfigFromKubeConfig(
+                case kind -> getKubernetesClientConfigFromKubeConfig(
                         KubeConfigUtils.parseKubeConfig(KubeConfigUtils.replaceServerInKubeconfig(containerAddress.getUrl(),
                                 getFileContentFromContainer(KIND_KUBECONFIG))));
-            } else if (image.contains("k8s.gcr.io/kube-apiserver") ||
-                    image.contains("registry.k8s.io/kube-apiserver")) {
-                return getKubernetesClientConfigFromKubeConfig(getKubeconfigFromApiContainer(containerAddress.getUrl()));
-            }
-
-            // this can happen only if the user manually start
-            // a DEV_SERVICE_LABEL labeled container with an invalid image name
-            throw new RuntimeException("The container with the label '" + DEV_SERVICE_LABEL
-                    + "' is not compatible with Dev Services for Kubernetes. Stop it or disable Dev Services for Kubernetes.");
+                case api_only ->
+                        getKubernetesClientConfigFromKubeConfig(getKubeconfigFromApiContainer(containerAddress.getUrl()));
+                default -> throw new RuntimeException("The container with the label '" + DEV_SERVICE_LABEL
+                        + "' is not compatible with Dev Services for Kubernetes. Stop it or disable Dev Services for Kubernetes.");
+            };
```

Comment:
Please don't use Java 17 language flavors in existing code, as it makes backports hard

Reply:
Time has passed and I think we are now fine with it as I don't expect us to backport this to 3.2.

Reply:
Makes sense